src/redux/middleware/sync.js

```diff
@@ -0,0 +1,58 @@
+import {
+  ACTIONS as LBRY_REDUX_ACTIONS,
+  makeSelectIsFollowingTag,
+  selectFollowedTags,
+} from 'lbry-redux';
+import Lbryio from 'lbryio';
+import * as ACTIONS from 'constants/action_types';
+import { selectSubscriptions } from 'redux/selectors/subscriptions';
+
+const persistShape = {
+  version: '0',
+  shared: {},
+};
+
+export function userStateSyncMiddleware() {
+  return ({ getState }) => next => action => {
+    if (
+      action.type === ACTIONS.CHANNEL_SUBSCRIBE ||
+      action.type === ACTIONS.CHANNEL_UNSUBSCRIBE ||
+      action.type === LBRY_REDUX_ACTIONS.TOGGLE_TAG_FOLLOW
+    ) {
+      const newShape = { ...persistShape };
+      const state = getState();
+      const subscriptions = selectSubscriptions(state).map(({ uri }) => uri);
+      const tags = selectFollowedTags(state);
+      newShape.shared.subscriptions = subscriptions;
+      newShape.shared.tags = tags;
+
+      const { uri } = action.data;
+
+      if (action.type === ACTIONS.CHANNEL_SUBSCRIBE) {
+        const newSubscriptions = subscriptions.slice();
+        newSubscriptions.push(uri);
+        newShape.shared.subscriptions = newSubscriptions;
+      } else if (action.type === ACTIONS.CHANNEL_UNSUBSCRIBE) {
+        let newSubscriptions = subscriptions.slice();
+        newSubscriptions = newSubscriptions.filter(subscribedUri => subscribedUri !== uri);
+        newShape.shared.subscriptions = newSubscriptions;
+      } else {
```

Comment:
I would keep this check explicit. You could also consider making this a `switch`.

Also, is there any way to have this fire after any reducers fire, so that you do not have to repeat the logic for each action that modifies the state?

I understand why you do not want a separate reducer for tracking shared state, but if you can have this function fire after reducers update redux state, then this logic becomes a lot simpler now and moving forward. All this function would need to do is re-select all user state and compare this state to the last previous known state.